src/posts/2020-09-19-how-to-write-accessible-forms.md

```diff
@@ -0,0 +1,154 @@
+
+Here are some  practical tips on how to write accessible forms.
+
+
+## Always label your forms
+
+Labels are a great way to get started with writing accessible forms. They are vital for helping people who use [screen readers](https://en.wikipedia.org/wiki/Screen_reader) and other kinds of assistive technology determine what label text go with what input field.
+
+To associate a `<label>` with an `<input>`, set the identifier(ID) in the `<input>` to correspond with the `for` attribute of the `<label>`. This is the generally accepted approach.
+```html
+<form>
+  <label for="firstName">First Name</label>
+  <input type="text" id="firstName" name="firtName"/>
+</form>
+```
+
+
+**Note**: It is adviceful not to nest `<input>` directly inside `<label>` because some assistive devices (such as [Dragon NaturallySpeaking](https://en.wikipedia.org/wiki/Dragon_NaturallySpeaking)) do not support it.
+
+
+## Highlight input element on focus
+
+Form insertion [carets](https://developer.mozilla.org/en-US/docs/Glossary/caret#:~:text=A%20caret%20sometimes%20called%20a,is%20called%20the%20insertion%20point.) can be difficult to track. A simple solution is to highlight `<input>` on [focus](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus). This way, people do not have to guess where they are on the form. 
+
+![A form with an input field highlighted in blue.](../img/posts/2020-09-19-how-to-write-accessible-forms/form-field-on-focus.png)
+
+```css
+input:focus {
+  outline: 2px solid royalblue;
+  box-shadow: 1px 1px 8px 1px royalblue;
+}
+```
+
+## Break long forms into smaller sections
+
+Filling long forms can be stressful. This is largely true for almost everyone but particularity for people  with [AD/HD](https://www.cdc.gov/ncbddd/adhd/facts.html). Such forms can lead to an increase in [Bounce rate](https://en.wikipedia.org/wiki/Bounce_rate) which isn't what anyone wants. A simple solution is to break them into smaller sections. This helps make long forms less daunting and easier to understand as it is provided in bits.
+
+The following principles apply for multi-step forms:
+
+- Repeat overall instruction on every page.
+- The forms should be split into logical groups.
+- The forms should inform the person about the progress thay are making.
+
+**Note**: There are no hard rules that defines a long form. However, you should consider if the form can be broken down into different logical sections.
+
+## Provide error messages
+
+When people encounter errors, they may need help to determine what's wrong and why. A common way forms indicate invalid fields is to change the `<input>` border color to red which is not ideal. The reason for this is because a change in color alone is not exposed to assistive technologies, so indicating errors in such ways will not inform the person of the error.

```

Comment:
Might also want to add that relying on color alone may also not work for someone who is red/green colorblind.